source/common/quic/client_connection_factory_impl.cc

```diff
@@ -43,12 +43,13 @@ std::shared_ptr<quic::QuicCryptoClientConfig> PersistentQuicInfoImpl::cryptoConf
 PersistentQuicInfoImpl::PersistentQuicInfoImpl(
     Event::Dispatcher& dispatcher, Network::TransportSocketFactory& transport_socket_factory,
     TimeSource& time_source, Network::Address::InstanceConstSharedPtr server_addr,
-    const quic::QuicConfig& quic_config, uint32_t buffer_limit)
+    const quic::QuicConfig& quic_config, uint32_t buffer_limit,
+    const envoy::config::core::v3::QuicProtocolOptions& protocol_config)
     : conn_helper_(dispatcher), alarm_factory_(dispatcher, *conn_helper_.GetClock()),
       server_id_{getConfig(transport_socket_factory).serverNameIndication(),
                  static_cast<uint16_t>(server_addr->ip()->port()), false},
       transport_socket_factory_(transport_socket_factory), time_source_(time_source),
-      quic_config_(quic_config), buffer_limit_(buffer_limit) {
+      quic_config_(quic_config), buffer_limit_(buffer_limit), protocol_config_(protocol_config) {
```

Comment:
I don't think you need to plumb through protoco_config in this way. EnvoyQuicClientSession::setHttp3Options() already gets cluster().http3Options() during client codec creation.

Reply:
Great tip! done removing the redundant protocol_config